app/controllers/api/databases.php

```diff
@@ -2664,14 +2664,21 @@ function updateAttribute(
 
         $database = Authorization::skip(fn() => $dbForProject->getDocument('databases', $databaseId));
 
-        if ($database->isEmpty() || (!$database->getAttribute('enabled') && $mode !== APP_MODE_ADMIN)) {
-            throw new Exception(Exception::DATABASE_NOT_FOUND);
+        $isAPIKey = Auth::isAppUser(Authorization::getRoles());
+        $isPrivilegedUser = Auth::isPrivilegedUser(Authorization::getRoles());
+        $isAdminMode = $mode === APP_MODE_ADMIN;
+        $isConsole = $isAdminMode && $isPrivilegedUser;
+
+        if ($database->isEmpty() || !$database->getAttribute('enabled')) {
+            if (!$isConsole && !$isAPIKey) {
+                throw new Exception(Exception::DATABASE_NOT_FOUND);
+            }
         }
```

Comment:
I wonder if this logic is correct 🧐. If the database doesn't exist, but I'm accessing this with an API key or as a Console User, this won't throw an exception, and it will continue to try and fetch the collection, causing unwanted/unexpected behavior.